### ISSUE TITLE
dalfox as a library 

### ISSUE THREAD

**Issue by lucmichalski** [hahwul/dalfox#235]
Hi @hahwul ,

Hope you are all well !

It would be awesome to use dalfox as a library like that:

```go
package main

import (
  "fmt"
  "time"

  dalfox "github.com/hahwul/dalfox/v2/lib"
)

func main() {
  opts := dalfox.Options{Worker: 100, ...}
  targeter := dalfox.NewScan(dalfox.Target{
    Options: opts,
    Method: "GET",
    URL:    "http://localhost:9100/",
  })
  scanner := targeter.Run()

  fmt.Printf("results: %s\n", scanner.Data)
}
```

Is it something that we can expect on the roadmap ?

Cheers,
Luc Michalski

**Comment by hahwul** [comment]
Hi @lucmichalski 
Yes, that's right! What I'm ultimately after is the support to library of scanner. This project have a cool contributors, but I'm noob and I don't have much time, so I don't know how soon I can support to the lib 😭

I need to roll up our sleeves. even if it's boring, please wait a little bit! 😎🚀

**Comment by lucmichalski** [comment]
lol you are crazy dude :-) completamente loco

On a more serious note, it would be really useful to contribute to dalfox if bundled as library, maybe go plugins ^^

**Comment by hahwul** [comment]
The part about the scan and model, structure be implemented to some extent and included in `v2.4.0`. (beta feature)

**Comment by hahwul** [comment]
@lucmichalski 
Sample of code
```go
package main 

import (
	"fmt"

	dalfox "github.com/hahwul/dalfox/v2/lib"
)

func main() {
	opt := dalfox.Options{
		Cookie:     "ABCD=1234",
	}
	result, err := dalfox.NewScan(dalfox.Target{
		URL:     "https://xss-game.appspot.com/level1/frame",
		Method:  "GET",
		Options: opt,
	})
	if err != {
		fmt.Println(err)
	} else {
		fmt.Println(result)
	}
}
```
result from `NewScan()` is interface > Result struct
https://github.com/hahwul/dalfox/blob/ea1a4482d97a5acd7e36f077482f968527a61393/lib/interface.go#L44
https://github.com/hahwul/dalfox/blob/ea1a4482d97a5acd7e36f077482f968527a61393/pkg/model/result.go#L8
